packages/edit-site/src/components/global-styles/global-styles-provider.js

```diff
@@ -45,22 +45,35 @@ const cleanEmptyObject = ( object ) => {
 };
 
 function useGlobalStylesUserConfig() {
-	const { globalStylesId, settings, styles } = useSelect( ( select ) => {
-		const _globalStylesId =
-			select( coreStore ).__experimentalGetCurrentGlobalStylesId();
-		const record = _globalStylesId
-			? select( coreStore ).getEditedEntityRecord(
-					'root',
-					'globalStyles',
-					_globalStylesId
-			  )
-			: undefined;
-		return {
-			globalStylesId: _globalStylesId,
-			settings: record?.settings,
-			styles: record?.styles,
-		};
-	}, [] );
+	const { globalStylesId, hasGlobalStyles, settings, styles } = useSelect(
+		( select ) => {
+			const { getEditedEntityRecord, getEntityRecords } =
+				select( coreStore );
+			const _globalStylesId =
+				select( coreStore ).__experimentalGetCurrentGlobalStylesId();
+			const record = _globalStylesId
+				? getEditedEntityRecord(
+						'root',
+						'globalStyles',
+						_globalStylesId
+				  )
+				: undefined;
+			const activeThemes = getEntityRecords( 'root', 'theme', {
+				status: 'active',
+			} );
+
+			return {
+				globalStylesId: _globalStylesId,
+				hasGlobalStyles:
+					!! activeThemes?.[ 0 ]?._links?.[
+						'wp:user-global-styles'
+					] ?? false,
```

Comment:
After recent changes, this link won't be available if a theme doesn't support `theme.json`. See https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php#L376-L380.